From 8c3d53f4fb5c745a1cf2126ff52188064fbab725 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Wed, 3 Oct 2018 22:28:40 +0900
Subject: [PATCH 18/28] nsis: Add uninstallation check

TODO: Should we run uninstaller again if it seems to be failed?
---
 nsis/gvim.nsi | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index 75c085c12..b360089ac 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -268,8 +268,21 @@ Section "$(str_section_old_ver)" id_section_old_ver
 	# run the install program to check for already installed versions
 	SetOutPath $TEMP
 	File /oname=install.exe ${VIMSRC}\installw32.exe
-	nsExec::Exec "$TEMP\install.exe -uninstall-check"
-	Pop $3
+	${Do}
+	  nsExec::Exec "$TEMP\install.exe -uninstall-check"
+	  Pop $3
+
+	  call CheckOldVim
+	  Pop $3
+	  ${If} $3 == ""
+	    ${ExitDo}
+	  ${Else}
+	    # It seems that the old version is still remaining.
+	    # TODO: Should we show a warning and run the uninstaller again?
+
+	    ${ExitDo}	# Just ignore for now.
+	  ${EndIf}
+	${Loop}
 	Delete $TEMP\install.exe
 	Delete $TEMP\vimini.ini   # install.exe creates this, but we don't need it.
 
-- 
2.17.0

